src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        AUTOMATED,
+        LIMITED,
+        ALL,
+        FORK,
+        FINALIZED
+    }
+
+    ReportingState private reportingState = ReportingState.AUTOMATED;
+
+    // CONSIDER: change the payoutNumerator/payoutDenominator to use fixed point numbers instead of integers; PRO: some people find fixed point decimal values easier to grok; CON: rounding errors can occur and it is easier to screw up the math if you don't handle fixed point values correctly
+    uint256 private payoutDenominator;
+    uint256 private feePerEthInAttoeth;
+
+    // CONSIDER: we really don't need these
+    int256 private maxDisplayPrice;
+    int256 private minDisplayPrice;
+
+    // CONSIDER: figure out approprate values for these
+    uint256 private constant AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT = 11 * 10**20;
+    uint256 private constant LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**21;
+    uint256 private constant ALL_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**22;
+
+    uint256 private constant MIN_NUM_OUTCOMES = 2;
```

Comment:
Style: I'm generally not a fan of constants that are only used once/in one place.  I am a fan of creating variables when a name helps make the code more readable though, and it is possible that you were running into stack size issues which is why these were extracted.  In some cases, I feel like it is obvious and the name doesn't add value, in which case I would inline (like `require(numOutcomes >= 2)` already conveys that 2 is the minimum number of outcomes.  Naming the `2` doesn't really add any value IMO.